### ISSUE TITLE
import util

### ISSUE THREAD

**Issue by Boraly** [taikamurmeli/Doc2VecC_python#2]
Hey,

I cant find a module names "util" in doc2vecc.py, do you mean "utils"? Or is it a custom module that is not included in the repo?

Thank you,
Boraly

**Comment by sjsarsa** [comment]
Hi!

My bad, thanks for pointing this out. "util" was a custom module that originally wasn't supposed to be needed in the public repo. I added the dependency in the latest commit cb78ba95f33276fcd9fd1621c5566bdf16c9d339.

Does doc2vecc.py work for you now?

**Comment by Boraly** [comment]
Hey,

I will try it out later. Thanks for the commit.
Another "mistake" I found was at doc2vecc.py at line 76 "self.alpha = .025 if cbow else .05".
It has to be the opposite way like you discribe at the params:
":param alpha: The starting learning rate; default is 0.025 for skip-gram and 0.05 for CBOW" 
If I am not mistaken it will be 0.025 when cbow is True, else 0.05. 

Thanks in advance,
Boraly

**Comment by sjsarsa** [comment]
Great, let me know how it works out.

Thanks for pointing out the default value mix up, it looks like you are correct. Now that I look at the code again more carefully after the years, it seems I'd better take a thorough look at the code (this'll have to wait for next week though).

**Comment by Boraly** [comment]
Hey, 

doc2vecc.py is working! But I compared the results of the python version with the wrapper of the original code and it seems off:
`(array([[ 5, 41, 78, 37, 19]]), array([[1.        , 0.9760333 , 0.9714191 , 0.9690429 , 0.96645737]],
      dtype=float32))
(array([[  5, 144, 482, 436, 271]]), array([[1.        , 0.95579517, 0.95314723, 0.95248544, 0.946822  ]],
      dtype=float32))`
I calculated the most similar documents for the document 5. At the top you can see the results of the python code and at the bottom the wrapper. I used the same hyperparameters and same corpus. I found, that the vectors of the python code are all much closer together. If I calculate the cosine similarity of the top 100 documents the lowest it gets is 0.917. For the wrapper its 0.816.
Comparing it to the "normal" Doc2Vec Algo its much closer together, too. Here I get the following results with the same hyperparameters:
`(array([[  5, 231, 144,  39, 271]]), array([[1.        , 0.9001148 , 0.8840287 , 0.8789093 , 0.86690176]],
      dtype=float32))`
,whereby the cosine similarity of the 100th document is 0.641.

Do you know if its normal, that the vectors of Doc2VecC are much closer then with Doc2Vec? And do you maybe know why?

**Comment by sjsarsa** [comment]
Great! I fixed the default alpha mess up in doc2vecc.py. I cleaned up the code a bit too.

 I wasn't able to pinpoint why the wrapper produces different vectors but the pymodule version didn't at least appear worse with some minor testing.
 
 My first bet for Doc2VecC vecs being closer is that it converges faster during training. At least with some simple test data, I got similarly close vectors for Doc2Vec by increasing the number of training epochs.
